WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -460,42 +460,39 @@ def submitJobs(self):
     # From here on we assume we are going to (try to) submit some pilots
     self.log.debug("Going to try to submit some pilots")
 
-    queues = self.queueDict.keys()
-    random.shuffle(queues)
-    self.log.verbose("Queues treated: %s" % ','.join(queues))
```

Comment:
This makes always the same order in which sites are treated for job submission. Not sure this is a good idea to drop it if even subsequent code is slightly nicer

Reply:
If the order should be randomized, insert a random order into the ordereddict, https://docs.python.org/2/library/collections.html#collections.OrderedDict